### PR TITLE
Use long for running deviation squared sum

### PR DESCRIPTION
**Pull request by vwbaker**
There was an instance of an overflow when storing the running deviation
squared sum as a double in b/198279407. Switching it to an unsigned long
while doing the running calculation allows us to accommodate for larger
numbers and doesn't sacrifice precision as it can't be a decimal until
it gets cast.

Bug: http://b/198279407

### CODE REVIEW

File: src/ClientData/CaptureData.cpp [hunk 1]
```diff
@@ -129,16 +127,18 @@ void CaptureData::OnCaptureComplete(
...
-          stats.set_variance_ns(stats.variance_ns() + deviation * deviation);
+          id_to_sum_of_deviations_squared[timer_info.function_id()] += deviation * deviation;
         }
       }
     }
   }
 
-  for (auto& pair : functions_stats_) {
-    FunctionStats& stats = pair.second;
+  for (auto& pair : id_to_sum_of_deviations_squared) {
```
Comment (pierricgimmig) [inline src/ClientData/CaptureData.cpp:136]: nit. structured bindings can make the code more legible : `for (auto& [id, sum_of_deviation_squared] : id_to_sum_of_deviations_squared)`

File: src/ClientData/CaptureData.cpp [hunk 2]
```diff
@@ -113,10 +114,7 @@ void CaptureData::AddFunctionStats(uint64_t instrumented_function_id,
 void CaptureData::OnCaptureComplete(
     const std::vector<const orbit_client_data::TimerChain*>& chains) {
   // Recalculate standard deviation as the running calculation may have introduced error.
-  for (auto& pair : functions_stats_) {
-    FunctionStats& stats = pair.second;
-    stats.set_variance_ns(0);
-  }
+  std::unordered_map<int, unsigned long> id_to_sum_of_deviations_squared;
```
Comment (pierricgimmig) [inline src/ClientData/CaptureData.cpp:117]: We prefer `absl::flat_hash_map` to `std::unordered_map`, it's a drop-in replacement with better performance.

**Review by pierricgimmig — APPROVED** [review]
Hi Tori, thanks for fixing the bug! Lgtm, but please see minor comments. Cheers

File: src/ClientData/CaptureData.cpp (hunk 1 above)
Comment (vwbaker) [reply src/ClientData/CaptureData.cpp]: done. thanks!

File: src/ClientData/CaptureData.cpp (hunk 2 above)
Comment (vwbaker) [reply src/ClientData/CaptureData.cpp]: done. thanks!

**Pull request merged**